core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -71,6 +71,8 @@ statement
         RENAME COLUMN (IF EXISTS)? from=identifier TO to=identifier    #renameColumn
     | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
         DROP COLUMN (IF EXISTS)? column=qualifiedName                  #dropColumn
+    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
+        ALTER COLUMN columnName=identifier alterColumnAction           #alterColumn
```

Comment:
> alterColumnAction

a  DROP COLUMN, RENAME COLUMN etc would be an alterColumnAction
we don't do that for these, so let's not do that for SET DATA TYPE (yet)
if we want to introduce alterColumnAction in the grammar, let's do this as a follow-up PR that we can discuss separately.

Reply:
Sorry, I'm not sure about the expected change from this comment. Do you recommend inlining `alterColumnAction` and rename `alterColumn ` to `setColumnType`? 

The reason why I wrote this way is this is the granularity of SQL spec and it will ease to add a new action.
```
<alter table action> ::=
    <add column definition> 
  | <alter column definition> 
  | <drop column definition> 
  | <add table constraint definition> 
  | <alter table constraint definition> 
  | <drop table constraint definition> 
  | <add table period definition> 
  | <drop table period definition> 
  | <add system versioning clause> 
  | <drop system versioning clause> 
```

Reply:
I think @findepi  means that you should simplify a bit the grammar since we don't currently have multiple options to alter a column.

```
    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
        ALTER COLUMN columnName=identifier SET DATA TYPE type          #alterColumn
```

I'd modify `#alterColumn` to `#alterColumnType`.

Reply:
yes, this is what i meant

Reply:
Updated.